src/plugins/orangehrmRecruitmentPlugin/Api/InterviewAttachmentAPI.php

```diff
@@ -0,0 +1,282 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Recruitment\Api;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CrudEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Entity\Interview;
+use OrangeHRM\Entity\InterviewAttachment;
+use OrangeHRM\Recruitment\Api\Model\InterviewAttachmentModel;
+use OrangeHRM\Recruitment\Traits\Service\RecruitmentAttachmentServiceTrait;
+
+class InterviewAttachmentAPI extends Endpoint implements CrudEndpoint
+{
+    use RecruitmentAttachmentServiceTrait;
+
+    public const PARAMETER_COMMENT = 'comment';
+    public const PARAMETER_INTERVIEW_ID = 'interviewId';
+    public const PARAMETER_ATTACHMENT_ID = 'attachmentId';
+    public const PARAMETER_ATTACHMENT = 'attachment';
+    public const PARAMETER_CURRENT_ATTACHMENT = 'currentAttachment';
+
+    public const PARAM_RULE_FILE_NAME_MAX_LENGTH = 200;
+    public const PARAM_RULE_CURRENT_ATTACHMENT_MAX_LENGTH = 16;
+    public const PARAM_RULE_COMMENT_MAX_LENGTH = 255;
+
+    public const INTERVIEW_ATTACHMENT_REPLACE_CURRENT = 'replaceCurrent';
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $interviewId = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            self::PARAMETER_INTERVIEW_ID
+        );
+        $attachments = $this->getRecruitmentAttachmentService()
```

Comment:
better to handle usual getAll with search filter param object :relaxed:

Reply:
done